templates/drone/index.ts

```diff
@@ -0,0 +1,106 @@
+import { 
+  Services,
+  createTemplate,
+  randomString 
+} from "~templates-utils";
+
+export default createTemplate({
+  name: "Drone",
+  meta: {
+    description: "Drone is a Container-Native, Continuous Delivery Platform",
+    changeLog: [{ date: "2022-08-03", description: "first release" }],
+    links: [
+      { label: "Website", url: "https://drone.io/" },
+      {
+        label: "Documentation",
+        url: "https://github.com/harness/drone#setup-documentation",
+      },
+      { label: "Github", url: "https://github.com/harness/drone" },
+    ],
+    contributors: [
+      { name: "Ivan Ryan", url: "https://github.com/ivanonpc-22" },
+    ],
+  },
+  schema: {
+    type: "object",
+    required: ["projectName", "domain", "serviceName", "clientID", "clientSecret"],
+    properties: {
+      projectName: {
+        type: "string",
+        title: "Project Name",
+      },
+      domain: {
+        type: "string",
+        title: "Domain",
+      },
+      serviceName: {
+        type: "string",
+        title: "App Service Name",
+        default: "drone",
+      },
+      clientID: {
+        type: "string",
+        title: "GitHub Oauth Client ID",
+        default: "ID",
+      },
+      clientSecret: {
+        type: "string",
+        title: "GitHub Oauth Client Secret",
+        default: "Secret",
+      },
+      RPCprotocol: {
+        type: "string",
+        title: "RPC Protocol",
+        default: "https",
+        oneOf: [
+          { enum: ["https"], title: "https" },
+          { enum: ["http"], title: "http" },
+        ],
+      },
+    },
+  } as const,
+    generate({
+      projectName,
+      domain,
+      serviceName,
+      clientID,
+      clientSecret,
+      RPCsecret = randomString(16),
```

Comment:
this has no property, move it below as a constant

`const rpcSecret = randomString(16);`